dart/common/Resource.hpp

```diff
@@ -75,6 +75,12 @@ class Resource
   /// \param[in] _count Number of elements, each of _size bytes.
   /// \note This method has the same API as the standard fread function.
   virtual std::size_t read(void *_buffer, std::size_t _size, std::size_t _count) = 0;
+
+  /// Reads all remaining data from this resource, and returns it as a string.
+  ///
+  /// \return The string retrieved from the resource.
+  /// \throw std::runtime_error when failed to read sucessfully.
+  std::string readAll();
```

Comment:
I suggest marking this `virtual`. Some classes may be able to implement this more efficiently than the generic implementation used in the base class.